src/main/java/tuteez/ui/DisplayCard.java

```diff
@@ -50,23 +53,60 @@ public DisplayCard(Optional<Person> lastViewedPerson) {
         if (lastViewedPerson.isPresent()) {
             name.setText(person.getName().fullName);
             phone.setText(person.getPhone().value);
-            telegram.setText(person.getTelegramUsername().telegramUsername);
+            setTelegramUsernameText(person);
             address.setText(person.getAddress().value);
             email.setText(person.getEmail().value);
-
-            tags.getChildren().clear();
             person.getTags().stream()
                     .sorted(Comparator.comparing(tag -> tag.tagName))
-                    .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
-
-            remarks.getChildren().clear();
-            person.getRemarkList().getRemarks().stream()
-                    .sorted(Comparator.comparing(remark -> remark.toString()))
-                    .forEach(remark -> remarks.getChildren().add(new Label(remark.toString())));
+                    .forEach(tag -> displayTags.getChildren().add(new Label(tag.tagName)));
+            IntStream.range(0, person.getRemarkList().getRemarks().size())
+                    .forEach(i -> {
+                        String remark = person.getRemarkList().getRemarks().get(i).toString();
+                        displayRemarks.getChildren().add(new Label((i + 1) + ". " + remark));
+                    });
+            setDisplayLessons(person);
+        }
+    }
 
-            lessons.getChildren().clear();
-            person.getLessons().stream()
-                    .forEach(lesson -> lessons.getChildren().add(new Label(lesson.getDayAndTime())));
+    /**
+     * Sets the Telegram username text for the provided person.
+     *
+     * This method checks if the provided person has a valid, non-empty Telegram username.
+     * If a username is available, it updates the display text with the username prefixed by "@" and
+     * makes the Telegram field visible. If no valid username is found, the Telegram field is hidden.
+     *
+     * @param person the Person object whose Telegram username is to be displayed
+     */
+    private void setTelegramUsernameText(Person person) {
+        TelegramUsername username = person.getTelegramUsername();
+        if (username != null && username.telegramUsername != null && !username.telegramUsername.isEmpty()) {
```

Comment:
Apologies but is there a reason where so many checks? It feels like all of these checks should be doing the same thing. Perhaps we could run through what is happening in our next huddle as I might not understand the full picture ;)